src/web/src/components/SearchProvider.tsx

```diff
@@ -1,40 +1,65 @@
 import { createContext, ReactNode, useState, useEffect, FormEvent } from 'react';
 import { useRouter } from 'next/router';
 
-type FilterProp = {
-  filter: 'post' | 'author' | string;
-};
-
 export interface SearchContextInterface {
-  text: string;
-  textParam: string;
-  filter: FilterProp['filter'];
+  post: string;
+  author: string;
+  from: string;
+  to: string;
+  title: string;
   showHelp: boolean;
   toggleHelp: (value: boolean) => void;
-  onTextChange: (value: string) => void;
-  onFilterChange: (value: FilterProp['filter']) => void;
+  onPostChange: (value: string) => void;
+  onAuthorChange: (value: string) => void;
+  onStartDateChange: (value: string) => void;
+  onEndDateChange: (value: string) => void;
   onSubmitHandler: (value: FormEvent) => void;
+  onTitleChange: (value: string) => void;
 }
 
 const SearchContext = createContext<SearchContextInterface>({
-  text: '',
-  textParam: '',
-  filter: 'post',
+  post: '',
+  author: '',
+  from: '',
+  to: '',
+  title: '',
   showHelp: true,
   toggleHelp() {
     throw new Error('This context must be wrapped inside SearchProvider');
   },
-  onTextChange() {
+  onPostChange() {
+    throw new Error('This context must be wrapped inside SearchProvider');
+  },
+  onAuthorChange() {
+    throw new Error('This context must be wrapped inside SearchProvider');
+  },
+  onStartDateChange() {
+    throw new Error('This context must be wrapped inside SearchProvider');
+  },
+  onEndDateChange() {
     throw new Error('This context must be wrapped inside SearchProvider');
   },
-  onFilterChange() {
+  onTitleChange() {
     throw new Error('This context must be wrapped inside SearchProvider');
   },
   onSubmitHandler() {
     throw new Error('This context must be wrapped inside SearchProvider');
   },
 });
 
+// Returns the string in the filter accordingly after loading results.
+function getFilter(filter: any) {
```

Comment:
Never use any in TS, this is why we use it instead of JS. Assuming it's always going to be a string, we can specify it as such